src/Symfony/Component/Routing/Loader/YamlFileLoader.php

```diff
@@ -127,6 +127,24 @@ protected function parseRoute(RouteCollection $collection, $name, array $config,
         $methods = isset($config['methods']) ? $config['methods'] : array();
         $condition = isset($config['condition']) ? $config['condition'] : null;
 
+        if (isset($requirements['_method'])) {
+            if (0 === count($methods)) {
+                $methods = explode('|', $requirements['_method']);
```

Comment:
@xabbuh I think this should be done onlu when ``$config['methods']`` is not set at all, not when having an empty array. This way, we would not overwrite it in case someone defines an empty array explicitly (which is a weird case though)